fix. Look for validation rules for bundles in app/Resources/config/ .…

volodyawork opened the pull request:
| Q | A |
| --- | --- |
| Branch? | 2.8 |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | yes/no (I don't know) |
| Deprecations? | no |
| Tests pass? | yes/no (Tested manually) |
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any |
| License | MIT |
| Doc PR | reference to the documentation PR, if any |

Look for validation rules for bundles in app/Resources/config/ .  Use  if  need overwrite default rules

volodyawork committed:
fix. Look for validation rules for bundles in app/Resources/config/ . Use  if  need overwrite default rules

sstok commented:
The usual way of overwriting validator Class constraints is to use a different Validation Groups.

Ps. Fill-in in the pull-request template 👍 use eg. `yes` or `no` not `yes/no` as this breaks @carsonbot 😄

tgalopin commented:
The issue here is to override validation from third-party bundles I think.

tgalopin commented on src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:
```diff
@@ -822,18 +822,28 @@ private function getValidatorMappingFiles(ContainerBuilder $container)
         foreach ($bundles as $bundle) {
             $reflection = new \ReflectionClass($bundle);
             $dirname = dirname($reflection->getFileName());
+            $overwriteDir = $container->getParameter('kernel.root_dir') . '/Resources/' . $reflection->getShortName();
```
You shouldn't put space in concatenations.

volodyawork committed:
fix. delete space in concatenations

volodyawork committed:
fix. delete space in concatenations

volodyawork commented:
@tgalopin  , I deleted spaces in concatenations.

@sstok , I changed pull-request template. Sorry, it is my first PR. Using different Validation Groups is not easiest way in every bundle. Ex. if we can't set Validation Groups in config file.

stof commented:
A validation file can define validation rules for any class. They are not bundle-specific. So I don't see the need to allow ``app/Resources/<bundle>/config/validation``. We should just allow ``app/Resources/validation`` or something like that (as done for translations btw).
Btw, allowing a global bundle-specific path without supporting the standard file looks weird.

And what is weird in your case is that creating ``app/Resources/<bundle>/config/validation/`` will disable loading of *all* files of the bundle, even if you overwrite only one of them.

tgalopin commented:
I think the issue with having a single app/Resources/validation file is that we will need to merge this file configuration with the other bundles one, which is much more complicated. If we use a per-bundle system, it simply override the bundle configuration.

stof commented:
@tgalopin we already allows to load all files inside a ``validation/`` folder in bundles, so the global config should allow the same if we add it.

tgalopin commented:
Hum ok. I think that's not the point of the PR though, but I get your point.

volodyawork commented:
@stof , @tgalopin  I agree with @stof about global validation. I can push https://gist.github.com/volodyawork/1dec9a151fee420caf51d8527d0f650d if we need that feature.
But it is change to structure of "Resources" folder, because we don't have "validation" folder in "Resources" folder of Bundle ( http://symfony.com/doc/2.8/bundles/best_practices.html#directory-structure ) .
What should I do?

stof commented:
@volodyawork it is inside ``Resources/config`` for bundles

volodyawork commented:
@stof, I know. 
Should we look for files in **app/Resources/config/validation** or **app/Resources/validation**?

nicolas-grekas commented:
Closing as there is no more activity, and the location of files changed significantly with Flex so this should be rethought I suppose. Feel free to reopen.
